internal/controller/controllers/infraenv_controller.go

```diff
@@ -718,44 +720,60 @@ func generateStaticNetworkConfigDownloadURL(baseURL string, infraEnvId string, a
 
 func (r *InfraEnvReconciler) updateInfraEnvStatus(
 	ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv, internalInfraEnv *common.InfraEnv) (ctrl.Result, error) {
-
-	osImage, err := r.OsImages.GetOsImageOrLatest(internalInfraEnv.OpenshiftVersion, internalInfraEnv.CPUArchitecture)
-	if err != nil {
-		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
-	}
-
-	bootArtifactURLs, err := imageservice.GetBootArtifactURLs(r.ImageServiceBaseURL, internalInfraEnv.ID.String(), osImage, r.InsecureIPXEURLs)
-	if err != nil {
-		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
-	}
-	infraEnv.Status.BootArtifacts.KernelURL = bootArtifactURLs.KernelURL
-	infraEnv.Status.BootArtifacts.RootfsURL = bootArtifactURLs.RootFSURL
-
+	var err error
+	var bootArtifactURLs *imageservice.BootArtifactURLs
 	var isoUpdated bool
-	if infraEnv.Status.ISODownloadURL != internalInfraEnv.DownloadURL {
-		log.Infof("ISODownloadURL changed from %s to %s", infraEnv.Status.ISODownloadURL, internalInfraEnv.DownloadURL)
-		infraEnv.Status.ISODownloadURL = internalInfraEnv.DownloadURL
+
+	// Update CreatedTime if no images are to be generated
+	if !r.ImageServiceEnabled && infraEnv.Status.CreatedTime == nil {
 		imageCreatedAt := metav1.NewTime(time.Time(internalInfraEnv.GeneratedAt))
 		infraEnv.Status.CreatedTime = &imageCreatedAt
 		isoUpdated = true
 	}
 
+	// Only update URLs if image service is enabled
+	if r.ImageServiceEnabled {
+		var osImage *models.OsImage
+		osImage, err = r.OsImages.GetOsImageOrLatest(internalInfraEnv.OpenshiftVersion, internalInfraEnv.CPUArchitecture)
+		if err != nil {
+			return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+		}
+
+		bootArtifactURLs, err = imageservice.GetBootArtifactURLs(r.ImageServiceBaseURL, internalInfraEnv.ID.String(), osImage, r.InsecureIPXEURLs)
+		if err != nil {
+			return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+		}
+		infraEnv.Status.BootArtifacts.KernelURL = bootArtifactURLs.KernelURL
+		infraEnv.Status.BootArtifacts.RootfsURL = bootArtifactURLs.RootFSURL
+
+		if infraEnv.Status.ISODownloadURL != internalInfraEnv.DownloadURL {
+			log.Infof("ISODownloadURL changed from %s to %s", infraEnv.Status.ISODownloadURL, internalInfraEnv.DownloadURL)
+			infraEnv.Status.ISODownloadURL = internalInfraEnv.DownloadURL
+			// Update CreatedTime when image is created (URL changed)
+			imageCreatedAt := metav1.NewTime(time.Time(internalInfraEnv.GeneratedAt))
+			infraEnv.Status.CreatedTime = &imageCreatedAt
```

Comment:
It feels like we could deduplicate this with the ones above.

Maybe update the created time if the url changed or we're disabled without a created time then just copy over the URL down here unconditionally and log if `isoUpdated`?